app/src/prod/java/com/andela/art/login/LoginActivity.java

```diff
@@ -0,0 +1,212 @@
+package com.andela.art.login;
+
+import android.app.ProgressDialog;
+import android.content.Intent;
+import android.databinding.DataBindingUtil;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.andela.art.R;
+import com.andela.art.root.ApplicationComponent;
+import com.andela.art.databinding.ActivityLoginBinding;
+import com.andela.art.login.injection.DaggerLoginComponent;
+import com.andela.art.login.injection.LoginModule;
+import com.andela.art.root.ApplicationModule;
+import com.andela.art.securitydashboard.presentation.SecurityDashboardActivity;
+import com.andela.art.root.ArtApplication;
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.common.api.Status;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthCredential;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.GoogleAuthProvider;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+/**
+ * LoginActivity handles the login of user into the application.
+ */
+public class LoginActivity extends AppCompatActivity implements SecurityEmailsView {
+
+    @Inject GoogleSignInClient mGoogleSignInClient;
+    @Inject SecurityEmailsPresenter securityEmailsPresenter;
+    FirebaseAuth mAuth;
+    private static final int RC_SIGN_IN = 2;
+    static final String TAG = "LoginActivity";
+    FirebaseAuth.AuthStateListener mAuthListener;
+    // A progress dialog to display when the user is connecting in
+    // case there is a delay in any of the dialogs being ready.
+    ProgressDialog mConnectionProgressDialog;
+    Intent dashboard;
+
+    public List<String> allowedEmailAddresses = new ArrayList();
+
+
+    // Custom error code for unauthorized GMail addresses
```

Comment:
@zmwas These are more of subtitles than comments. In this case the variable name is descriptive enough, no need to prepend it with a comment. Let's only use inline comments whenever they are absolutely necessary.